Add support for libvirt and GCE

geertj opened the pull request:
Add support for building libvirt and GCE appliance.

These are two separate commits but in one PR. The commits step on each other so it's easier to have it in one PR.

geertj commented:
@simaishi can you have a look at this?

Fryguy commented on kickstarts/partials/packages/includes.ks.erb:
```diff
@@ -91,6 +91,8 @@ npm
 ovirt-guest-agent
 glusterfs-fuse
+<% elsif @target == "libvirt" %>
+qemu-guest-agent
```
Minor but in a followup PR can we sort this and use a case statement? cc @simaishi

geertj replied:
Happy to do that. Still earning Ruby :)

Fryguy commented on scripts/target.rb:
```diff
@@ -8,7 +8,9 @@ class Target
       'azure'     => ImagefactoryMetadata.new('hyperv', nil, 'vhd'),
-      'vagrant'   => ImagefactoryMetadata.new('vsphere', 'vagrant-virtualbox', 'box')
+      'vagrant'   => ImagefactoryMetadata.new('vsphere', 'vagrant-virtualbox', 'box'),
+      'libvirt'   => ImagefactoryMetadata.new('openstack-kvm', nil, 'qc2'),
```
Isn't this the same as the openstack image?  Would it be simpler to just rename the openstack image or copy it instead of a full build?

geertj replied:
@Fryguy the target name drives certain conditional statements in the kickstarts. So while the definition looks identical, it will result in a different appliance.

simaishi commented on scripts/target.rb:
```diff
@@ -8,7 +8,9 @@ class Target
-      'vagrant'   => ImagefactoryMetadata.new('vsphere', 'vagrant-virtualbox', 'box')
+      'vagrant'   => ImagefactoryMetadata.new('vsphere', 'vagrant-virtualbox', 'box'),
+      'libvirt'   => ImagefactoryMetadata.new('openstack-kvm', nil, 'qc2'),
+      'gce'       => ImagefactoryMetadata.new('gce', nil, 'tgz'),
```
@geertj Can you change the extension to`.tar.gz`?  That's GCE's requirement and disk can't be created from `.tgz` image.

simaishi commented on README.md:
```diff
@@ -97,8 +97,8 @@ Below are instructions on installing and configuring a virtual machine to genera
-  * Clone imagefactory as /build/imagefactory (requires version 1.1.9-3 or
-    later for command line parameter passing and hyperv target support):
+  * Clone imagefactory as /build/imagefactory (requires git tag f13dc65
+    or later for google compute engine support):
```
The GCE support also has the same symlink problem, just like HyperV.  Can you please add a note about manually creating symlink for GCE?

geertj commented:
@simaishi I've added the note about the symlink and changed the file extension to .tar.gz. The reason that I didn't run into the file extension issue is that I used the recently merged capability in imagefactory to push directly to GCE. That code adds a `.tar.gz`  file extension automatically. My next PR will enable this in manageiq-appliance-build.

@Fryguy let me know if my comment above on why libvirt is different from openstack makes sense. As you suggest let's switch the if/elsif to a case in a follow up PR. I have some more changes coming in that area which which I can fold this in.

Anything else before this can be merged?

simaishi commented:
Looks good for GCE, thanks @geertj !

geertj commented:
@simaishi thanks!

@

geertj commented:
@simaishi thanks!

@Fryguy @jrafanie can this be merged please? @simaishi checked GCE and I personally checked the libvirt piece. Also the libvirt piece is trivial, it only adds the `qemu-guest-agent` package.

jrafanie commented:
Merged @geertj, thanks!
